File: src/main/java/de/ayont/lpc/channels/RangeChannel.java
package de.ayont.lpc.channels;

import de.ayont.lpc.LPC;
import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.entity.Player;

public class RangeChannel extends BaseChannel {

    private final double range;

    public RangeChannel(LPC plugin, String id, ConfigurationSection section) {
        super(plugin, id, section);
        this.range = section.getDouble("range", 100.0);
    }

    @Override
    public boolean canRead(Player player, Player sender) {
        if (!player.getWorld().equals(sender.getWorld())) return false;
        return player.getLocation().distanceSquared(sender.getLocation()) <= range * range;
    }
}

File: src/main/kotlin/de/ayont/lpc/channels/RangeChannel.kt
package de.ayont.lpc.channels

import de.ayont.lpc.LPC
import org.bukkit.configuration.ConfigurationSection
import org.bukkit.entity.Player

class RangeChannel(plugin: LPC, id: String, section: ConfigurationSection) : BaseChannel(plugin, id, section) {

    private val range: Double = section.getDouble("range", 100.0)

    override fun canRead(player: Player, sender: Player): Boolean {
        if (player.world != sender.world) return false
        return player.location.distanceSquared(sender.location) <= range * range
    }
}
